### PR TITLE
Add architecture docs README, readings about ADRs

### PR DESCRIPTION
**Pull request by KevinMGranger**
This pulls out the architecture doc portion of #834, since ADRs have been handled separately.

It also adds the helpful readings about how/why to use ADRs from that PR.

### CODE REVIEW

File: docs/Development.md [hunk 1]
```diff
@@ -73,6 +77,14 @@ Within the Pelorus GitHub fork directory:
 
        To ensure the implementation's pull request is merged **after** the ADR's PR it is recommended to add `Depends on: #ADR_PULL_REQUEST_NUMBER` as the first comment on the implementation PRs. Please follow the [dpulls](https://www.dpulls.com/docs#/dependencies) for more information.
 
+## Internal Architecture Docs
```
[the remark this answers is not in the document]
Comment (KevinMGranger) [reply docs/Development.md:80]: That makes sense to me!

However, I think docs for the deserialization and config loading parts should be kept where they are. They're essentially "sub-libraries"-- they'd be external libraries we'd import if someone else was supporting them :)

[the remark this answers is not in the document]
Comment (KevinMGranger) [reply docs/Development.md:80]: Hmm, there's both "here's how this thing works internally" and "here's why we chose this". I wonder what the best name is.

[the remark this answers is not in the document]
Comment (KevinMGranger) [reply docs/Development.md:80]: Actually, now I want to move parts of the deserialization and config loading docs into this too. I'll make some changes and see what you think.

[the remark this answers is not in the document]
Comment (KevinMGranger) [reply docs/Development.md:80]: Updated, let me know what you think!

[the remark this answers is not in the document]
Comment (mateusoliveira43) [reply docs/Development.md:80]: I like!

But now this section is wrong (new lines 80-87). Removing it is already LGTM to me

[the remark this answers is not in the document]
Comment (KevinMGranger) [reply docs/Development.md]: Removed!

**Review by mateusoliveira43 — APPROVED** [review]

**Pull request merged**